esmtools/carbon.py

```diff
@@ -125,3 +123,180 @@ def potential_pco2(t_insitu, pco2_insitu):
+    # sensitivities at each grid cell.
+    # TODO: Add keyword for sliding mean, as in N year chunks of time to
+    # account for trends.
+    DIC = ds['dissicos']
+    ALK = ds['talkos']
+    SALT = ds['sos']
+    pCO2 = ds['spco2']
+
+    buffer_factor = dict()
+    buffer_factor['ALK'] = -ALK**2 / ((2 * DIC - ALK) * (ALK - DIC))
+    buffer_factor['DIC'] = (3*ALK*DIC - 2*DIC**2) / \
+                           ((2 * DIC - ALK) * (ALK - DIC))
+    # Compute sensitivities
+    sensitivity = dict()
+    sensitivity['tos'] = 0.0423
+    sensitivity['sos'] = 1 / SALT
+    sensitivity['talkos'] = (1 / ALK) * buffer_factor['ALK']
+    sensitivity['dissicos'] = (1 / DIC) * buffer_factor['DIC']
+    sensitivity = xr.Dataset(sensitivity) * pCO2
+    return sensitivity
+
+
+# TODO: adapt for CESM and MPI output.
+@check_xarray([0, 1])
+def spco2_decomposition_index(ds_terms, index, detrend=False,
+                              deseasonalize=False, plot=False,
+                              sliding_window=None,
+                              **plot_kwargs):
+    """Decompose oceanic surface pco2 in a first order Taylor-expansion.
+
+    Reference:
+    * Lovenduski, Nicole S., Nicolas Gruber, Scott C. Doney, and Ivan D. Lima.
+        “Enhanced CO2 Outgassing in the Southern Ocean from a Positive Phase of
+        the Southern Annular Mode.” Global Biogeochemical Cycles 21, no. 2
+        (2007). https://doi.org/10/fpv2wt.
+
+    Args:
+        ds (xr.Dataset): containing cmorized variables:
+                            spco2 [ppm]: pCO2,ocean at ocean surface
+                            talkos[mmol m-3]: Alkalinity at ocean surface
+                            dissicos[mmol m-3]: DIC at ocean surface
+                            tos [C] : temperature at ocean surface
+                            sos [psu] : salinity at ocean surface
+        index (xr.object): Climate index to regress onto.
+        detrend (bool): Whether to detrend time series prior to regression.
+        deseasonalize (bool): Whether to deseasonalize time series prior to
+                              regression.
+        plot (bool): quick plot. Defaults to False.
+        sliding_window (int): Number of years to apply sliding window to for
+                              calculation.
+        **plot_kwargs (type): `**plot_kwargs`.
+
+    Returns:
+        terms_in_pCO2_units (xr.Dataset): terms of spco2 decomposition,
+                                          if `not plot`
+
+    """
+    def regression_against_index(ds, index, psig=None):
+        terms = dict()
+        for term in ds.data_vars:
+            if term != 'spco2':
+                reg = linregress(index, ds[term], psig=psig)
+                terms[term] = reg['slope']
+        terms = xr.Dataset(terms)
+        return terms
+
+    pco2_sensitivity = spco2_sensitivity(ds_terms)
+    if detrend:
+        ds_terms_anomaly = rm_trend(ds_terms, dim='time')
```

Comment:
should we maybe change to `rm_poly` and add `order` to the func args? In case we want to calculate this for a period from 1850 to 2100 which is non-linear in spco2?

Reply:
Added